com.unity.render-pipelines.high-definition/Editor/Material/Nature/SpeedTree8MaterialUpgrader.cs

```diff
@@ -0,0 +1,146 @@
+        public SpeedTree8MaterialUpgrader(string sourceShaderName, string destShaderName)
+        {
+            RenameShader(sourceShaderName, destShaderName, SpeedTree8MaterialFinalizer);
+        }
+
+        private static void SpeedTree8MaterialFinalizer(Material material)
+        {
+            // Run after upgrading from built-in SpeedTree8 shader.
+            if (material.name.Contains("Billboard"))
+            {
+                material.SetFloat("EFFECT_BILLBOARD", 1);
+                material.EnableKeyword("EFFECT_BILLBOARD");
+            }
+            HDShaderUtils.ResetMaterialKeywords(material);
+        }
+
+        internal static void SetST8MaterialKeywords(Material material, int windInt = -1)
+        {
+            // When a speedtree8 is imported or a speedtree8 material is upgraded, internal floats are set correctly
+            // and can be read to reapply keywords after resetting them via HDShaderUtils.ResetMaterialKeywords.
+            // However, when a material inspector is rendered for the first time, it does not have the correct
+            // property values but does have the correct keywords.
+            // In both cases, a keyword is only ever set if it is correct.
+            // So in order to sync keywords with exposed properties, we check first if a relevant keyword is set.
+            // Only if it is not set do we check the property to get the value.
+            // For wind in particular, we only ever pass in non-negative windInt when we intend to set it to that value.
+            // So in summary:
+            // Priority order of property-controlled keywords that require syncing:
+            // EFFECT_BILLBOARD: enabled keyword > property value
+            // _WINDQUALITY*: valid windInt > enabled keyword > property value
+
+            if (material != null)
+            {
+                if (material.shader.name.Equals("HDRP/Nature/SpeedTree8"))
+                {
+                    material.EnableKeyword("_ALPHATEST_ON");
+                    material.EnableKeyword("ENABLE_WIND");
+                    // Assumes all ST8 materials are double-sided.
+                    material.EnableKeyword("_DOUBLESIDED_ON");
+
+                    if (!WindIntValid(windInt))
+                    {
+                        windInt = GetWindQualityFromKeywords(material.shaderKeywords);
+                        if (!WindIntValid(windInt))
+                            windInt = material.HasFloat("_WindQuality") ? (int)material.GetFloat("_WindQuality") : 0;
+                        if (!WindIntValid(windInt))
+                            windInt = 0;
+                    }
+
+                    material.EnableKeyword(WindQualityString[windInt]);
+                    material.SetFloat("_WindQuality", windInt); // A legacy float used in native code to apply wind data
+                    material.SetFloat("_WINDQUALITY", windInt); // The actual name of the keyword enum for the shadergraph
```

Comment:
I am not sure this works to set a keyword. If you are relying on the fact that the _WINDQUALITY property has a shaderlab attribute, I think you need to call
```MaterialEditor.ApplyMaterialPropertyDrawers(material)```
in order for the keywords to be regenerated
Maybe it's called somewhere else for you but I am not sure